### PR TITLE
[jiralert] Setup chart

### PR DESCRIPTION
**Pull request by jkroepke**
Signed-off-by: Jan-Otto Kröpke <jok@cloudeteer.de>

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes https://github.com/prometheus-community/jiralert/issues/104

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by jkroepke** [comment]
Test chart job failes, because no default receivers are defined. This results into a CrashLoopBackOff. Should I add some non working default receivers to get the lint job working?

**Comment by monotek** [comment]
> Test chart job failes, because no default receivers are defined. This results into a CrashLoopBackOff. Should I add some non working default receivers to get the lint job working?

Yes, chart should be able to start the container with the default config. Maybe also some test values in the ci directory would do the trick.

File: charts/jiralert/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "jiralert.fullname" . }}
+  namespace: {{ template "jiralert.namespace" . }}
+  labels:
+    {{- include "jiralert.labels" . | nindent 4 }}
+data:
+  jiralert.yml: {{ tpl (.Values.config | toYaml) . | b64enc }}
+  jiralert.tmpl: {{ tpl .Values.issueTemplate . | b64enc }}
```
Comment (monotek) [inline charts/jiralert/templates/secret.yaml:10]: it should also be possible to get these values from exisitng secrets.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/jiralert/values.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+podAnnotations: {}
+
+# -- jiralert pods' Security Context.
+podSecurityContext: {}
+# fsGroup: 2000
+
+# Container security context
+securityContext:
+  runAsUser: 1001
+  runAsGroup: 1
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply charts/jiralert/values.yaml]: Done

File: charts/jiralert/templates/secret.yaml (hunk 1 above)
Comment (jkroepke) [reply charts/jiralert/templates/secret.yaml:11]: Done

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -7,6 +7,7 @@
 * @prometheus-community/helm-charts-admins
 
 /charts/alertmanager/ @monotek @naseemkullah
+/charts/jiralert/ @jkroepke
```
Comment (zanhsieh) [inline .github/CODEOWNERS:10]: More than one maintainer would be better.
```suggestion
/charts/jiralert/ @jkroepke @zanhsieh 
```

File: charts/jiralert/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+appVersion: "1.2"
+home: "https://github.com/prometheus-community/jiralert"
+keywords:
+  - monitoring
+  - prometheus
+  - alerting
+  - jiralert
+maintainers:
+  - name: jkroepke
+    email: github@jkroepke.de
```
Comment (zanhsieh) [inline charts/jiralert/Chart.yaml:15]:
```suggestion
  - name: jkroepke
    email: github@jkroepke.de
  - name: MH
    email: zanhsieh@gmail.com
```

File: charts/jiralert/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "jiralert.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+        {{- with .Values.podAnnotations }}
+          {{- toYaml . | nindent 8 }}
```
Comment (zanhsieh) [inline charts/jiralert/templates/deployment.yaml:18]:
```suggestion
        {{- toYaml . | nindent 8 }}
```

File: charts/jiralert/templates/ingress.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+{{- end }}
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  namespace: {{ template "jiralert.namespace" . }}
+  labels:
+    {{- include "jiralert.labels" . | nindent 4 }}
+    {{- if .Values.ingress.labels }}
+{{ toYaml .Values.ingress.labels | indent 4 }}
+    {{- end}}
```
Comment (zanhsieh) [inline charts/jiralert/templates/ingress.yaml:24]:
```suggestion
    {{- with .Values.ingress.labels }}
    {{- toYaml . | nindent 4 }}
    {{- end}}
```

File: charts/jiralert/templates/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (zanhsieh) [inline charts/jiralert/templates/ingress.yaml:15]: Could we move this to _helper.yaml? 

In this file:
```
apiVersion: {{template "jiralert.ingress.apiversion" . }}
```

In _helpers.tpl:
```
{{/*
Return the appropriate apiVersion for ingress.
*/}}
{{- define "jiralert.ingress.apiversion" -}}
{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
{{- print "networking.k8s.io/v1" -}}
{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
{{- print "networking.k8s.io/v1beta1" -}}
{{- else -}}
{{- print "extensions/v1beta1" -}}
{{- end }}
```

File: charts/jiralert/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            {{- toYaml .Values.readinessProbe | nindent 12 }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            - name: config
+              mountPath: "/config"
+              readOnly: true
+      serviceAccountName: {{ include "jiralert.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
```
Comment (zanhsieh) [inline charts/jiralert/templates/deployment.yaml:55]:
```suggestion
      {{- with .Values.podSecurityContext }}
      securityContext:
        {{- toYaml .Values.podSecurityContext | nindent 8 }}
      {{- end }}        
```

File: charts/jiralert/templates/ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
```
Comment (zanhsieh) [inline charts/jiralert/templates/ingress.yaml:42]:
```suggestion
  {{- with .Values.ingress.tls }}
  tls:
    {{- range . }}
    - hosts:
        {{- range .hosts }}
        - {{ . | quote }}
        {{- end }}
      secretName: {{ .secretName }}
    {{- end }}
  {{- end }}
```

File: charts/jiralert/values.yaml [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+  #      - chart-example.local
+
+issueTemplate: |
+  {{`{{ define "jira.priority" }}{{ if eq (index .Alerts 0).Labels.severity "critical" }}High{{ else if eq (index .Alerts 0).Labels.severity "warning" }}Medium{{ end }}{{ end }}
+  {{ define "jira.summary" }}[{{ .Status | toUpper }}{{ if eq .Status "firing" }}:{{ .Alerts.Firing | len }}{{ end }}] {{ .GroupLabels.SortedPairs.Values | join " " }}{{ end }}
+
+  {{ define "jira.description" }}{{ range .Alerts.Firing }}Labels:
+  {{ range .Labels.SortedPairs }} - {{ .Name }} = {{ .Value }}
+  {{ end }}
+                                                                       Annotations:
```
Comment (zanhsieh) [inline charts/jiralert/values.yaml:144]: Why so many whitespaces in front of `Annotations`?

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**